sharing/Readme_sharing.md

```diff
@@ -1,31 +1,145 @@
+
+- **who -** The sharing property or the rule can identify who the data is intended for. Different data repositories or organizations or public or a person can have different access level to the data as per the requirements of the data custodian. The different access groups will be identified through these rules, **shared_with_access_groups** that will be part of the schema in the **shared_data.csv** file. When the data custodian logs onto the web portal to use ODM tool for data sharing, they will have options to select for the access groups they wish to share the data with.
+
+- **description -** There will be a description for each sharing property or rule that will specify the license or permissions on how the data or specific data elements shared with a specific recipient can or can not be used.
+
+- **extensible -** Since the waste water surveilance is rapidly developing, it will be necessary to make the sharing process more adaptable and extensible for growing changes and processes that allows users to add more variables and rules in the schema for sharing.
+
+### Sharing schema and rules
+
+- The sharing schema defines the rules that will be used to selectively filter the data for the data custodians who want to share only specific part of their data with a specific data repository. The ODM will create the main schema in the **shared_data.csv** file which will be updated in each version. However, the ODM users will be able to extend or develop the schemas for their own specific applications.
+
+- Pandas is a library in python which provides features to do extensive data manipulations including conditional filtration based on different rules. These rules are provided by the data custodian in **shared_data.csv** file which is converted into a json python object which is then used with Pandas library to select variables based on those provided rules. Having the schema for sharing in a CSV format allows non-data scientists to develop their specific data sharing specifications or rules.
+
+- A sharing schema allows ODM users to provide consistent format for sharing data that can be used to filter data based on their specifications.
+
+Following are the rules in the version v1.2 that are part of **shared_data.csv** file:
+
+**is_excluded:** This rule will allow the user to exclude any variable from their data that they do not wish to share with any data repository or organizations at all. The rule will accept values as True/ False based on whether the variable can be shared or not shared.
+
+**shared_with_Public:** This rule will allow the user to exclude any variable that the data custodian may not wish to share with the general public based on True/ False values specified for that rule in the sharing schema.
```

Comment:
I suggest that we retire the historic sharing approach in WWMeasure. I am not aware of any user who is using those sharing rules. @jeandavidt  what do you think?

Reply:
I suggest that we are flexible and allow on sharing schema to be used for multiple end-users. We would use your approach of the prefix `share_with_...` to identify an end-user. Who the data is shared with is defined by the custodian. Meaning, any header with `share_with_` identifies that header as a sharing entity. 

i.e. `share_with_myFriend-A`, `share_with_myFriend-B`, 

A data custodian could also have multiple schemas if they wish. They could, for example, have a separate schema, `share_data_myFriend-A.csv`and `share_data_myFriend-B.csv`.

Reply:
Yes that will work, because the way code is structured, it will not be hard coded with whom they share. But what is essential is that the last part of the name of who they share with should be what they input into the web application form. This way code will figure out who they are sharing with as a different access group.

Reply:
Hmm. We may want to include a `sharing_with...` label and/or description with the schema. The label and description would be used with the web app and other uses.

`sharing_with_Friend-A`, `label_Friend-A`, `description_Friend-A`.

Reply:
should I write it as only one sharing rule in the specification document such as **sharing_with_label** where label is who they are sharing the data with. This will cut down on the number of sharing rules we have in specification document because the user will be able to make as many labels as they want when they work with the schema, so I can simply add it as a single generalized property or rule: "sharing_with_label** in the specification?